app/concepts/sage_network/client.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'openssl'
+require 'uri'
+require 'socket'
+
+module SageNetwork
+  # The client class to perform API calls
+  class Client
+    # Set the bearer token
+    def self.bearer_token=(token)
+      RequestStore.store[:lockstep_bearer_token] = token
+      RequestStore.store[:lockstep_api_key] = nil
+
+      true
+    end
+
+    # Get the bearer token
+    def self.bearer_token
+      RequestStore.store[:lockstep_bearer_token]
+    end
+
+    ##
+    # Construct a new Lockstep API client targeting the specified server.
+    #
+    # @param env [string] Either "sbx", "prd", or the URI of the server, ending in a slash (/)
+    def initialize(env)
+      @version = '2022.4.32.0'
+      @env = case env
+             when 'sbx'
+               'https://api.sbx.lockstep.io/'
+             when 'prd'
+               'https://api.lockstep.io/'
+             else
+               env
+             end
```

Comment:
This will be changed when the api urls are confirmed by Platform for each environment.